### ISSUE TITLE
proposal: repository for Linux version

### ISSUE THREAD

**Issue by edmundlaugasson** [sozi-projects/Sozi#513]
### Summary of your problem or feature request

As soon as Sozi would like to reach more users, a repository for Linux would be needed. This would allow to automate installation and update the software in a way, we know software is installed under Linux. Such software repository allows to dramatically increase also security to ensure, that software package comes from authentic source, that is digitally signed (GPG) and can be trusted.

### Version of Sozi and other relevant software

This applies to all versions of Sozi. As currently there is DEB-version available, an appropriate Debian repository for all Debian-based distros could be created.

### Steps to reproduce the problem

Every time new version of Sozi comes out, it needs to be manually installed. In larger scale it is overwhelming and not usable. Then organizations will create own repository in order to solve that issue.

### Expected behavior

Would expect [PPA](https://help.ubuntu.com/community/PPA) availability. The abbreviation stands for Personal Package Archive.

### Observed behavior

Currently manual installation only possible.

### Hints and solutions (optional)

* Packaging might be overwhelming and limited for certain Linux distros and its versions. Although DEB-package is very much appreciated.
* Therefore e.g. [Flatpak](https://flatpak.org/) might be also appealing packaging format, that allows to use Sozi in a convenient way on more distros. Flatpak has also [online overview](https://flathub.org/) of available packages.
* [AppImage](https://appimage.org/) might be even more appealing, but does not allow central installaton, updating automation. Although an AppImage can be made so, that it allows in-app updating, *.desktop automatic creation on first run. For personal scale it is enough, for organizations, I guess not.
* https://wiki.debian.org/DebianRepository/Setup might be good start for *.deb packages repository
* https://launchpad.net/ is quite famous free server for Personal Package Archives
* there are lots of relatively short tutorials available, e.g. https://www.tecmint.com/create-deb-pacakge-repository-in-ubuntu/ , https://rpmdeb.com/devops-articles/how-to-create-local-debian-repository/

**Comment by NedNet29** [comment]
I agree, I am personnally using Fedora and didn't mange to install Sozi with Alien utility, I had too much missing dependencies.
So I ended using it in a VM, either W10 or Linux Mint in order to use the provided "official" installers.
I wish a Fedora packager would port Sozi to Fedoras repositories (I'm not skilled enough to do this).

**Comment by aumouvantsillage** [comment]
@edmundlaugasson I totally agree with the motivations of you proposal.

In the history of Sozi, there have been several attempts to include Sozi in distributions: [in Debian](https://tracker.debian.org/pkg/sozi), [in a PPA for Ubuntu](https://launchpad.net/~sunab/+archive/ubuntu/sozi-release), [in Fedora](https://src.fedoraproject.org/rpms/inkscape-sozi), and [in the Archlinux User Repository (AUR)](https://aur.archlinux.org/packages/sozi/).

All these packages were created and maintained by volunteers. After Sozi 13, only the AUR continued to be updated. It is the only repository that offers the latest stable release.
I see at least two reasons for that:

* Sozi 13 was the last version that was implemented as an Inkscape extension. The following versions had a completely different architecture and build process, which required the package maintainers to restart from scratch.
* Sozi 15 and above are based on [electron](https://www.electronjs.org/). Sozi depends on several node.js packages, which implies fetching external sources during the build process. I heard that it was incompatible with Launchpad's PPAs but maybe it's no longer the case.

This is not a problem for Archlinux because the AUR does not actually contain a Sozi *package*: the repository only contains [a script](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=sozi) that builds Sozi on the user's computer. As a consequence, it does not suffer from the restrictions of Debian or Ubuntu.

I think two solutions can be considered for Linux:

* Since I believe that my Debian packages do not meet the criteria for inclusion in official repositories, Sozi could have its own alternative repository.
* Flatpak seems promising, but I need some help. It does not seem as straightforward as creating a Debian package.

@NedNet29 I have recently changed the configuration of the packaging tool. The next packages that I publish will have fewer dependencies, and possibly better compatibility with other distributions.

**Comment by NedNet29** [comment]
Thank's for the RPMs package : I'll test it, along with W10 & deb (on Mint)

**Comment by NedNet29** [comment]
Hi, I just installed v21.05 beta on Fedora 33 (downloaded RPM and used CLI "sudo dnf install sozi....rpm"), it worked OK.
Just a suggestion (I'll open a specific issue for that if you prefer) : I'm using Mate Desktop, and Sozi's launcher appeared in "accessories" category, I personaly expected it in "Graphics".
